Assets/ConsentManagementProvider/Scripts/model/common/UsnatConsent.cs

```diff
@@ -0,0 +1,126 @@
+        public bool applies;
+        public List<ConsentString> consentStrings;
+        public List<Consentable> vendors;
+        public List<Consentable> categories;
+		public StatusesUsnat statuses;
+
+        public UsnatConsent(
+#nullable enable
+                        string? uuid,
+                        bool applies,
+                        List<ConsentString> consentStrings,
+                        List<Consentable> vendors,
+                        List<Consentable> categories,
+						ConsentStatus consentStatus
+#nullable disable
+        ) {
+            this.uuid = uuid;
+            this.applies = applies;
+            this.consentStrings = consentStrings;
+            this.vendors = vendors;
+            this.categories = categories;
+            this.statuses = StatusesUsnat.collectData(consentStatus);
+        }
+        
+        public string ToFullString()
+        {
+            StringBuilder sb = new StringBuilder();
+
+            sb.AppendLine($"USNAT");
+
+            if(uuid != null)
+                sb.AppendLine($"UUID: {uuid}");
+            sb.AppendLine($"Applies: {applies}");
+
+            sb.AppendLine($"ConsentStrings:");
+            foreach (var _string in consentStrings)
+            {
+                sb.AppendLine($"    sectionId: {_string.sectionId}, sectionName: {_string.sectionName}, consentString: {_string.consentString}");
+            }
+            sb.AppendLine("Categories:");
+            foreach (var _consentable in categories)
+            {
+                sb.AppendLine($"    id:{_consentable.id}, consented:{_consentable.consented}");
+            }
+            sb.AppendLine("Vendors:");
+            foreach (var _consentable in vendors)
+            {
+                sb.AppendLine($"    id:{_consentable.id}, consented:{_consentable.consented}");
+            }
+            sb = statuses.ToFullString(sb);
+            return sb.ToString();
+        }
+    }
+
+    public class StatusesUsnat
+    {        
+        public bool? rejectedAny, consentedToAll, consentedToAny,
+            hasConsentData, sellStatus, shareStatus,
+            sensitiveDataStatus, gpcStatus;
+
+        internal static StatusesUsnat collectData(ConsentStatus status)
+        {
+            return new StatusesUsnat
+            {
+                rejectedAny = status.rejectedAny,
+                consentedToAll = status.consentedToAll,
+                consentedToAny = status.consentedToAny,
+                hasConsentData = status.hasConsentData,
+                sellStatus = status.granularStatus?.sellStatus,
+                shareStatus = status.granularStatus?.shareStatus,
+                sensitiveDataStatus = status.granularStatus?.sensitiveDataStatus,
+                gpcStatus = status.granularStatus?.gpcStatus
+            };
+        }
+
+        public StringBuilder ToFullString(StringBuilder sb)
+        {
+            sb.AppendLine("ConsentStatus:");
+            if(rejectedAny != null)
```

Comment:
I believe there is some kind of `switch` statement in C#, why don't we use that here?

Reply:
@bohdan-go-wombat It could be possible if we check the same thing all over again. 
In this scenario, we constantly check _different_ things in order to add them to `StringBuilder` (the tool that provides optimized string operations) to produce a huge string representation of an object to pretty-print it in the console. That's why we cannot use `switch` here. But thanks for the suggestion!